java/src/main/java/com/microsoft/onnxruntime/ONNXJavaType.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright © 2019, Oracle and/or its affiliates. All rights reserved.
+ */
+package com.microsoft.onnxruntime;
+
+import com.microsoft.onnxruntime.TensorInfo.ONNXTensorType;
+
+/**
+ * An enum over supported Java primitives (and String).
+ */
+public enum ONNXJavaType {
+    FLOAT(1, float.class, 4),
+    DOUBLE(2, double.class, 8),
+    INT8(3, byte.class, 1),
+    INT16(4, short.class, 2),
+    INT32(5, int.class, 4),
+    INT64(6, long.class, 8),
+    BOOL(7, boolean.class, 1),
+    STRING(8, String.class, 4),
+    UNKNOWN(0, Object.class, 0);
+
+    private static final ONNXJavaType[] values = new ONNXJavaType[9];
+
+    static {
+        for (ONNXJavaType ot : ONNXJavaType.values()) {
+            values[ot.value] = ot;
+        }
+    }
+
+    public final int value;
+    public final Class<?> clazz;
+    public final int size;
+
+    ONNXJavaType(int value, Class<?> clazz, int size) {
+        this.value = value;
+        this.clazz = clazz;
+        this.size = size;
+    }
+
+    /**
+     * Maps from an int in native land into an ONNXJavaType instance.
+     * @param value The value to lookup.
+     * @return The enum instance.
+     */
+    public static ONNXJavaType mapFromInt(int value) {
+        if ((value > 0) && (value < values.length)) {
+            return values[value];
+        } else {
+            return UNKNOWN;
+        }
+    }
+
+    /**
+     * Must match the values from ONNXUtil.c.
+     *
+     * @param onnxValue The value from ONNXUtil.c
+     * @return A JavaDataType instance representing the Java type
+     */
+    public static ONNXJavaType mapFromONNXTensorType(ONNXTensorType onnxValue) {
+        switch (onnxValue) {
+            case ONNX_TENSOR_ELEMENT_DATA_TYPE_UINT8:
+            case ONNX_TENSOR_ELEMENT_DATA_TYPE_INT8:
+                return ONNXJavaType.INT8;
+            case ONNX_TENSOR_ELEMENT_DATA_TYPE_UINT16:
```

Comment:
>case ONNX_TENSOR_ELEMENT_DATA_TYPE_UINT16: [](start = 12, length = 42)

We will have to decide on what to do with Tensors that contain unsigned types. Perhaps, the should not be allowed especially, when you created it from Java API.

Reply:
In maps I support the types that are specified as supported in the c API header, for unsigned types this is rather unfortunate but it does allow users to use models with unsigned types even if they have to contort their calling code.

Reply:
It's easy enough to support unsigned 8-bit and 16-bit types in Java with helper methods that pass values using `int` arguments. Here's an example of that using JavaCPP's indexer package:
http://bytedeco.org/news/2014/12/23/third-release/